### PR TITLE
Use code excerpts for /language/patterns page

### PR DESCRIPTION
**Pull request by johnpryan**
Contributes to https://github.com/dart-lang/site-www/issues/4828

### CODE REVIEW

**Comment by parlough** [comment]
Everyone prepared some code excerpts! 😊

Thank you!

File: src/language/patterns.md [hunk 1]
```diff
@@ -301,12 +317,17 @@ You need to validate it first to confirm its structure.
 
 Without patterns, validation is verbose:
 
+<?code-excerpt "misc/lib/language_tour/patterns/json.dart (json-2)"?>
 ```dart
+Object json = {};
```
Comment (parlough) [inline src/language/patterns.md:322]: Can you remove this line from the snippet?

File: src/language/patterns.md [hunk 2]
```diff
@@ -301,12 +317,17 @@ You need to validate it first to confirm its structure.
...
 if (json is Map<String, dynamic> &&
-    json.length == 1 && json.containsKey('user')) {
-  var user = json[user];
-  if (user is List<dynamic> && user.length == 2 &&
-      user[0] is String && user[1] is int) {
+    json.length == 1 &&
```
Comment (parlough) [inline src/language/patterns.md:324]: I was going to say this is really hard to read but that's the point :)

**Review by parlough — APPROVED** [review]
Thanks for this!

I left one suggestion. Seems like you're getting `unused_local_variable` warnings too that you can probably just ignore for the whole file.

Also, not fully necessary in this PR, but could you move these examples to the new [`examples/language`](https://github.com/dart-lang/site-www/tree/main/examples/language) directory? I'm trying to get away from everything being nested deep under `misc`.

File: examples/misc/lib/language_tour/patterns/algebraic_datatypes.dart [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+class Circle implements Shape {
+  final double radius;
+  Circle(this.radius);
+}
+
+double calculateArea(Shape shape) => switch (shape) {
+      Square(length: var l) => l * l,
+      Circle(radius: var r) => math.pi * r * r
+    };
+// #enddocregion algebraic_datatypes
```
Comment (MaryaBelanger) [inline examples/misc/lib/language_tour/patterns/algebraic_datatypes.dart:20]: Did you get a bunch of errors on this? I've been trying to implement the same example for the Branches page and it's giving me all kinds of errors...

Comment (johnpryan) [reply examples/misc/lib/language_tour/patterns/algebraic_datatypes.dart:20]: I needed to import the math library to get it working:

```
import 'dart:math' as math;
```

**Review by parlough — APPROVED** [review]
Okay, I moved the examples to `/language` and added ignore comments for the lints breaking CI.

Thanks for working on this. I feel a lot better having these analyzed :D

**Comment by johnpryan** [comment]
Thanks for cleaning this up!

**Pull request merged**